app/controllers/foreman_inventory_upload/uploads_controller.rb

```diff
@@ -22,7 +22,9 @@ def download_file
 
     def enable_cloud_connector
       # Set the autoupload to true, since it's required by the feature.
-      Setting[:allow_auto_inventory_upload] = true
+      if Foreman.settings.find('allow_auto_inventory_upload')
```

Comment:
Would prefer all these `Foreman.settings.find` to be replaced by a wrapper method whose name shows its intent, e.g. 
```
def set_non_iop_setting(setting_name, value)
  if Foreman.settings.find(setting_name)
    Setting[setting_name] = value
  end
end
```

Reply:
Let's first try and hide this whole mess into the settings infrastructure as I have suggested.